src/components/Main/Results/DeterministicLinePlot.tsx

```diff
@@ -100,12 +100,12 @@ export function DeterministicLinePlot({
     caseCounts,
   )
 
-  const countObservations = {
-    cases: caseCounts?.filter((d) => d.cases).length ?? 0,
-    ICU: caseCounts?.filter((d) => d.icu).length ?? 0,
-    observedDeaths: caseCounts?.filter((d) => d.deaths).length ?? 0,
-    newCases: caseCounts?.filter((_0, i) => newEmpiricalCases[i]).length ?? 0,
-    hospitalized: caseCounts?.filter((d) => d.hospitalized).length ?? 0,
+  const hasObservations = {
+    [DATA_POINTS.ObservedCases]: caseCounts && caseCounts.some((d) => d.cases),
+    [DATA_POINTS.ObservedICU]: caseCounts && caseCounts.some((d) => d.icu),
+    [DATA_POINTS.ObservedDeaths]: caseCounts && caseCounts.some((d) => d.deaths),
+    [DATA_POINTS.ObservedNewCases]: newEmpiricalCases && newEmpiricalCases.some((d) => d),
+    [DATA_POINTS.ObservedHospitalized]: caseCounts && caseCounts.some((d) => d.hospitalized),
```

Comment:
Rename 'countObservations' to 'hasObservations', and simplify the logic to reflect the purpose: It measures presence, not magnitude.